partner_validate_vat/res_partner.py

```diff
@@ -24,47 +24,37 @@
 #
 #
 
-from openerp.osv import osv
-from openerp.tools.translate import _
+from openerp import models, api, _
+from openerp.exceptions import ValidationError
 
 
-class ResPartner(osv.Model):
+class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    def _check_vat_uniqueness(self, cr, uid, ids, context=None):
+    @api.multi
+    @api.constrains('vat')
```

Comment:
For this case it is not possible, since we must validate not only the field that is not repeated in table of the database, it is already valid that a parnet with parent.id have the same vat